app/controllers/page_list_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PageListController < ApplicationController
+  include CheckFormOrganisation
+
+  def edit
+    @form = Form.find(params[:form_id])
+    @pages = @form.pages
+    @mark_complete_form = Forms::MarkCompleteForm.new(form: @form).assign_form_values
+    @mark_complete_options = mark_complete_options
+  end
+
+  def update
+    @form = Form.find(params[:form_id])
+    @pages = @form.pages
+    @mark_complete_form = Forms::MarkCompleteForm.new(mark_complete_form_params)
+    @mark_complete_options = mark_complete_options
+
+    if @mark_complete_form.valid?
+      @form.question_section_completed = @mark_complete_form.mark_complete
+      if @form.save
+        redirect_to form_path(@form)
+      else
+        raise StandardError, "Save unsuccessful"
+      end
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  rescue StandardError => e
+    flash[:message] = e
+    render :edit, status: :unprocessable_entity
+  end
+
+private
+
+  def mark_complete_options
+    [OpenStruct.new(value: "true"), OpenStruct.new(value: "false")]
+  end
+
+  def mark_complete_form_params
+    params.require(:forms_mark_complete_form).permit(:mark_complete)
+  end
+end
```

Comment:
Could we maybe make this controller a bit more generic? I like the fact that the html for the component has been made generic but this controller feels very specific to pages. Maybe a "MarkSectionCompleted" controller?

Reply:
@aliuk2012 
I'm in two minds about this - while it would be nice to make it more generic the notion of marking the pages task as complete feels like it should be quite closely coupled to the pages list. I guess the way I was imagining it is that the MarkCompleteComponent might appear on multiple pages, but it would post back to different controllers depending on what route was passed into it.

Does that make sense? This might be somewhere where my lack of knowledge re: Rails patterns is letting me down, but I couldn't work out a way of separating out this logic cleanly.

Reply:
ok,  lets leave it the way you did it. If we find we need it for MarkSectionComplete for other tasks then we can revisit it. Rule of 3 and keeping things DRY :wink: